### DISCUSSION TITLE
Uso CDN Chamilo 1.11.26

### DISCUSSION THREAD

**Discussion by soporteLecciona** [chamilo/chamilo-lms#5967] [General] [answered]
Buenas,
en el archivo configuration.php, tenemos habilitada la CDN, con distintas extensiones, entre ellas .jpg y .jpeg.
Sin embargo, la imágen que hemos subido a la página principal de nuestro chamilo, así como las imágenes asociadas a los cursos, en lugar de peticionarlas la URL a la CDN que tenemos configurada, las sigue peticionando al servidor donde está alojado Chamilo. ¿hay alguna configuración extra para que se sustituya en la URL de estas imágenes la URI base de la CDN en vez de la URI del dominio?
Pongo configuración:
**configuration.php**
// Set the following setting to true to start using the CDN**
$_configuration['cdn_enable'] = true;
// The following setting will be ignored if the previous one is set to false
$_configuration['cdn'] = [
    // You can define several CDNs and split them by extensions
    // Replace the following by your full CDN URL, which should point to
    // your Chamilo's root directory. DO NOT INCLUDE a final slash! (won't work)
    'https://chamiloespana.b-cdn.net' => [
        '.css',
        '.js',
       '.jpg',
       '.jpeg',
       '.png',
       '.gif',
       '.avi',
       '.pdf',
    ],
    // copy the line above and modify following your needs
];
**URL CDN (los headers de CORS están configurados correctamente en BunnyCDN para las extensiones configuradas):**
https://chamiloespana.b-cdn.net

**URL** Imágenes en general e imágen página principal:
<img alt="" height="475" src="https://campus.aulaglobal.online/app/upload/users/1/1/my_files/principal_chamilo-min.jpg" width="800">

Gracias

**Comment by soporteLecciona** [comment]
Al pegar la URL de la imágen, me hasacado la imágen en sí en el foro. Pego de nuevo la URL de esta imágen:

```
<img alt="" height="475" src="https://campus.aulaglobal.online/app/upload/users/1/1/my_files/principal_chamilo-min.jpg" width="800">
```

**Comment by ywarnier** [comment, marked as answer]
Hola @soporteLecciona 

El CDN solo funciona con las imágenes (y librerías CSS/JS) estáticas que vienen incluidas en Chamilo, no las publicadas por usuarios.

Normalmente deberías ver en la fuente de tu página que carga los JS y CSS (por lo menos) desde `https://chamiloespana.b-cdn.net/` (o la URL que uses como CDN).

Fue un experimento (funcional) que no se llevó más allá, pero podría ser aumentado progresivamente para cubrir otros recursos a través de scripts como `main/document/download.php`, por ejemplo, el problema siendo que hay una validación de accesos a través de download.php que no podrías tener relegando la descarga inmediatamente al CDN (tendrías que pasar igual por download.php para verificar permisos, y luego mandar la descarga al CDN, lo cual anularía parte del beneficio de tener un CDN).
